### ISSUE TITLE
Update the oldest supported macOS from 10.12

### ISSUE THREAD

**Issue by mike-myers-tob** [osquery/osquery#7574]
# Blueprint

Proposing to update minimum supported macOS from 10.12 to 10.14 or 10.15.

Continuous Integration currently tests macOS builds of osquery against macOS 10.15. So we aren't testing these older versions (10.12 through 10.14) even though we say they are supported.

Updating the minimum SDK will allow osquery to support more modern C++ features.

@sharvilshah @Smjert @directionless

**Comment by directionless** [comment]
For reference:

| Version | Name | Released |
| ------- | ---- | -------|
| 10.12 | Sierra | September 20, 2016 |
| 10.13 | High Sierra | September 25, 2017 |
| 10.14 | Mojave |  September 24, 2018 |
| 10.15 | Catalina | October 7, 2019 |
| 11.0 | Big Sur |  November 12, 2020 |
| 12.0 | Monterey | October 25, 2021|

I think 10.13 is totally fine -- High Sierra was akin to a point release on Sierra. 

I'm on the fence about whether it's reasonable to bump to 10.14. I suspect it's pretty universal. but if we wanted 5 years that's 10.13

I think 10.15 is right out.

**Comment by mike-myers-tob** [comment]
Since I opened this issue, GitHub Actions has deprecated the 10.15 runner, so now macOS 10.12, 10.13, 10.14 and [10.15 cannot be tested in CI](https://github.com/osquery/osquery/pull/7700): only x86 macOS 11 is being tested currently. Hopefully soon we can have [tests running on macOS 12 as well](https://github.com/osquery/osquery/issues/7702). But with macOS 13 Ventura likely 6-8 weeks away, this seems like a good time to shift the support window forward to 10.14, which is the most recent dead macOS (see table above, I added the support end date column).
